pgrx v0.15.0 on Postgres 18 error[E0308]: mismatched types

From [this build](https://github.com/theory/pg-jsonschema-boon/actions/runs/15947818310/job/44984066077), I see these errors building against Postgres 18 on Debian:

```
error[E0308]: mismatched types
Error:    --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:273:22
    |
273 |             .replace(pgrx_executor_start)
    |              ------- ^^^^^^^^^^^^^^^^^^^ expected fn pointer, found fn item
    |              |
    |              arguments to this method are incorrect
    |
    = note: expected fn pointer `unsafe extern "C-unwind" fn(_, _) -> ()`
                  found fn item `unsafe extern "C-unwind" fn(_, _) -> bool {pgrx_executor_start}`
help: the return type of this call is `unsafe extern "C-unwind" fn(*mut pgrx_pg_sys::QueryDesc, i32) -> bool {pgrx_executor_start}` due to the type of the argument passed
   --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:272:35
    |
272 |           prev_executor_start_hook: pg_sys::ExecutorStart_hook
    |  ___________________________________^
273 | |             .replace(pgrx_executor_start)
    | |______________________-------------------^
    |                        |
    |                        this argument influences the return type of `replace`
note: method defined here
   --> /rustc/6b00bc3880198600130e1cf62b8f8a93494488cc/library/core/src/option.rs:1809:18

error[E0308]: mismatched types
Error:    --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:274:22
    |
274 |             .or(Some(pgrx_standard_executor_start_wrapper)),
    |                 ---- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected fn pointer, found fn item
    |                 |
    |                 arguments to this enum variant are incorrect
    |
    = note: expected fn pointer `unsafe extern "C-unwind" fn(_, _) -> ()`
                  found fn item `unsafe extern "C-unwind" fn(_, _) -> bool {pgrx_standard_executor_start_wrapper}`
help: the type constructed contains `unsafe extern "C-unwind" fn(*mut pgrx_pg_sys::QueryDesc, i32) -> bool {pgrx_standard_executor_start_wrapper}` due to the type of the argument passed
   --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:274:17
    |
274 |             .or(Some(pgrx_standard_executor_start_wrapper)),
    |                 ^^^^^------------------------------------^
    |                      |
    |                      this argument influences the type of `Some`
note: tuple variant defined here
   --> /rustc/6b00bc3880198600130e1cf62b8f8a93494488cc/library/core/src/option.rs:597:5

error[E0308]: mismatched types
Error:    --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:345:25
    |
345 |         HookResult::new(res)
    |         --------------- ^^^ expected `bool`, found `()`
    |         |
    |         arguments to this function are incorrect
    |
note: associated function defined here
   --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:43:12
    |
43  |     pub fn new(value: T) -> Self {
    |            ^^^ --------

error[E0308]: mismatched types
Error:    --> /usr/share/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pgrx-0.15.0/src/hooks.rs:734:5
    |
733 | ) -> bool {
    |      ---- expected `bool` because of return type
734 |     pg_sys::standard_ExecutorStart(query_desc, eflags)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `()`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `pgrx` (lib) due to 4 previous errors
Error: Process completed with exit code 1.
```

The error does not occur on Postgres 13-17.

--- a/pgrx-pg-config/src/lib.rs
+++ b/pgrx-pg-config/src/lib.rs
@@ -713,12 +713,12 @@ pub fn SUPPORTED_VERSIONS() -> Vec<PgVersion> {
         PgVersion::new(17, PgMinorVersion::Latest, None),
         PgVersion::new(
             18,
-            PgMinorVersion::Beta(1),
+            PgMinorVersion::Beta(2),
             Some(
                 Url::parse(
-                    "https://ftp.postgresql.org/pub/source/v18beta1/postgresql-18beta1.tar.bz2",
+                    "https://ftp.postgresql.org/pub/source/v18beta2/postgresql-18beta2.tar.bz2",
                 )
-                .expect("malformed pg18beta1 url"),
+                .expect("malformed pg18beta2 url"),
             ),
         ),
     ]

--- a/pgrx-tests/src/tests/hooks_tests.rs
+++ b/pgrx-tests/src/tests/hooks_tests.rs
@@ -37,30 +37,12 @@ mod tests {
                 prev_hook(error_data)
             }
 
-            #[cfg(any(
-                feature = "pg13",
-                feature = "pg14",
-                feature = "pg15",
-                feature = "pg16",
-                feature = "pg17"
-            ))]
             fn executor_start(
                 &mut self,
                 query_desc: PgBox<QueryDesc>,
                 eflags: i32,
                 prev_hook: fn(PgBox<QueryDesc>, i32) -> HookResult<()>,
             ) -> HookResult<()> {
-                self.events += 1;
-                prev_hook(query_desc, eflags)
-            }
-
-            #[cfg(feature = "pg18")]
-            fn executor_start(
-                &mut self,
-                query_desc: PgBox<QueryDesc>,
-                eflags: i32,
-                prev_hook: fn(PgBox<QueryDesc>, i32) -> HookResult<bool>,
-            ) -> HookResult<bool> {
                 self.events += 1;
                 prev_hook(query_desc, eflags)
             }

--- a/pgrx/src/hooks.rs
+++ b/pgrx/src/hooks.rs
@@ -64,30 +64,12 @@ pub trait PgHooks {
     }
 
     /// Hook for plugins to get control in ExecutorStart()
-    #[cfg(any(
-        feature = "pg13",
-        feature = "pg14",
-        feature = "pg15",
-        feature = "pg16",
-        feature = "pg17"
-    ))]
     fn executor_start(
         &mut self,
         query_desc: PgBox<pg_sys::QueryDesc>,
         eflags: i32,
         prev_hook: fn(query_desc: PgBox<pg_sys::QueryDesc>, eflags: i32) -> HookResult<()>,
     ) -> HookResult<()> {
-        prev_hook(query_desc, eflags)
-    }
-
-    /// Hook for plugins to get control in ExecutorStart()
-    #[cfg(feature = "pg18")]
-    fn executor_start(
-        &mut self,
-        query_desc: PgBox<pg_sys::QueryDesc>,
-        eflags: i32,
-        prev_hook: fn(query_desc: PgBox<pg_sys::QueryDesc>, eflags: i32) -> HookResult<bool>,
-    ) -> HookResult<bool> {
         prev_hook(query_desc, eflags)
     }
 
@@ -307,13 +289,6 @@ pub unsafe fn register_hook(hook: &'static mut (dyn PgHooks)) {
     pg_sys::RegisterXactCallback(Some(xact_callback), std::ptr::null_mut());
 }
 
-#[cfg(any(
-    feature = "pg13",
-    feature = "pg14",
-    feature = "pg15",
-    feature = "pg16",
-    feature = "pg17"
-))]
 #[pg_guard]
 unsafe extern "C-unwind" fn pgrx_executor_start(query_desc: *mut pg_sys::QueryDesc, eflags: i32) {
     fn prev(query_desc: PgBox<pg_sys::QueryDesc>, eflags: i32) -> HookResult<()> {
@@ -327,26 +302,6 @@ unsafe extern "C-unwind" fn pgrx_executor_start(query_desc: *mut pg_sys::QueryDe
     }
     let hook = &mut HOOKS.as_mut().unwrap().current_hook;
     hook.executor_start(PgBox::from_pg(query_desc), eflags, prev);
-}
-
-#[cfg(feature = "pg18")]
-#[pg_guard]
-unsafe extern "C-unwind" fn pgrx_executor_start(
-    query_desc: *mut pg_sys::QueryDesc,
-    eflags: i32,
-) -> bool {
-    fn prev(query_desc: PgBox<pg_sys::QueryDesc>, eflags: i32) -> HookResult<bool> {
-        let res = unsafe {
-            (HOOKS.as_mut().unwrap().prev_executor_start_hook.as_ref().unwrap())(
-                query_desc.into_pg(),
-                eflags,
-            )
-        };
-        HookResult::new(res)
-    }
-    let hook = &mut HOOKS.as_mut().unwrap().current_hook;
-    hook.executor_start(PgBox::from_pg(query_desc), eflags, prev);
-    true
 }
 
 #[cfg(any(
@@ -710,27 +665,11 @@ unsafe extern "C-unwind" fn pgrx_emit_log(error_data: *mut pg_sys::ErrorData) {
     hook.emit_log(PgBox::from_pg(error_data), prev).inner
 }
 
-#[cfg(any(
-    feature = "pg13",
-    feature = "pg14",
-    feature = "pg15",
-    feature = "pg16",
-    feature = "pg17"
-))]
 #[pg_guard]
 unsafe extern "C-unwind" fn pgrx_standard_executor_start_wrapper(
     query_desc: *mut pg_sys::QueryDesc,
     eflags: i32,
 ) {
-    pg_sys::standard_ExecutorStart(query_desc, eflags)
-}
-
-#[cfg(feature = "pg18")]
-#[pg_guard]
-unsafe extern "C-unwind" fn pgrx_standard_executor_start_wrapper(
-    query_desc: *mut pg_sys::QueryDesc,
-    eflags: i32,
-) -> bool {
     pg_sys::standard_ExecutorStart(query_desc, eflags)
 }
 

